udata_front/theme/gouvfr/assets/js/components/ProducerSelector/ProducerSelector.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <div v-if="isAdmin()">
+    <MultiSelect
+      :required="true"
+      :minimumCharacterBeforeSuggest="2"
+      :placeholder="$t('Check the identity with which you want to publish')"
+      :searchPlaceholder="$t('Select an organization')"
+      suggestUrl="/organizations/suggest/"
+      :initialOptions="organizations"
+      :values="userOrganization"
+      @change="updateOrganization"
+    />
+  </div>
+  <div v-else-if="hasOrganizations">
+    <MultiSelect
+      :required="true"
+      :minimumCharacterBeforeSuggest="2"
+      :placeholder="$t('Check the identity with which you want to publish')"
+      :searchPlaceholder="$t('Select an organization')"
+      :initialOptions="organizations"
+      :values="userOrganization"
+      @change="updateOrganization"
+    />
+  </div>
+  <div v-else>
+    <div class="fr-col bg-contrast-grey text-align-center fr-p-2v">
+      <p class="fr-text--md fr-text--bold fr-mb-n3v">You belong to no organization</p>
+      <p class="fr-text--sm fr-text--bold fr-mb-n4v">You publish in your own name</p>
+      <p class="fr-text--sm">We advise you to publish under an organization if it's a professional activity</p>
+      <div class="fr-grid-row fr-grid-row--middle fr-pb-3v">
+        <div class="fr-col-6">
+          <button class="fr-btn fr-btn--secondary fr-btn--secondary-grey-500" @click="">
+            {{ $t("Join an organization") }}
+          </button>
+        </div>
+        <div class="fr-col-6">
+          <button class="fr-btn" @click="">
+            {{ $t("Create an organization") }}
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, toValue, onMounted } from 'vue';
+import MultiSelect from '../../components/MultiSelect/MultiSelect.vue';
+import type { User, Organization } from '@etalab/data.gouv.fr-components';
+
+const props = defineProps<{user: User}>();
+
+const userOrganization = ref<Organization>();
+const organizations = ref<Array<Organization>>([]);
+const emit = defineEmits(['update:organization']);
+
+const isAdmin = () => {
```

Comment:
can you use a computed instead ? This would match the way you did it for `hasOrganizations`